cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGenerator.scala

```diff
@@ -0,0 +1,654 @@
+          case s: String =>
+            s"""$fieldExpr in ( "${filter.values.mkString("\",\"")}" )"""
+        }
+      case Relation.contains =>
+        val sb = new StringBuilder
+        val wordsArr = ArrayBuffer[String]()
+        filter.values.foreach(w => wordsArr += "%" + w + "%")
+        for (i <- 0 until (wordsArr.length - 1)){
+          sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(i)}' and ")
+        }
+        sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(wordsArr.length - 1)}'")
+        sb.toString()
+      case Relation.matches => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"""$fieldExpr="${values(0)}""""
+      }
+      case _ => {
+        s"$fieldExpr ${filter.relation} ${filter.values(0)}"
+      }
+    }
+  }
+
+  protected def parseAggregateFunc(aggr: AggregateStatement,
+                                   fieldExpr: String): String = {
+    def aggFuncExpr(aggFunc: String): String = {
+      if (aggr.field.name.equals("*")) {
+        s"$aggFunc($groupVar)"
+      } else {
+        s"$aggFunc($fieldExpr)"
+      }
+    }
+
+    aggr.func match {
+      case topK: TopK => ???
+      case DistinctCount => ???
+      case _ =>
+        aggFuncExpr(getAggregateStr(aggr.func))
+    }
+  }
+
+  private def parseFilter(filters: Seq[FilterStatement], exprMap: Map[String, FieldExpr], unnestTestStrs: Seq[String], queryBuilder: StringBuilder): ParsedResult = {
+    if (filters.isEmpty && unnestTestStrs.isEmpty) {
+      ParsedResult(Seq.empty, exprMap)
+    } else {
+      val filterStrs = filters.map { filter =>
+        filter.field.dataType match {
+          case js: DataType.Json.type if (filter.field.name.contains(".")) =>
+            parseFilterRelation(filter, sourceVar + '.' + filter.field.name.replaceFirst("\\.", "->\"\\$.") + '"')
+          case _ =>
+            parseFilterRelation(filter, exprMap(filter.field.name).refExpr)
+        }
+      }
+      val filterStr = (unnestTestStrs ++ filterStrs).mkString("where ", " and ", "")
+      appendIfNotEmpty(queryBuilder, filterStr)
+      ParsedResult(Seq.empty, exprMap)
+    }
+  }
+
+  protected def parseGeoCell(scale: Double, fieldExpr: String, dataType: DataType.Value): String = ???
+
+  private def parseUnnest(unnest: Seq[UnnestStatement],
+                          exprMap: Map[String, FieldExpr], queryBuilder: StringBuilder): ParsedResult = {
+    val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
+    val unnestTestStrs = new ListBuffer[String]
+    val unnestStr = unnest.zipWithIndex.map {
+      case (unnest, id) =>
+        val expr = exprMap(unnest.field.name)
+        val newExpr = s"${quote}hashtags$quote"
+        producedExprs += (unnest.as.name -> FieldExpr(newExpr, newExpr))
+        if (unnest.field.isOptional) {
+          unnestTestStrs += s"${expr.refExpr} is not null"
+        }
+        s"" // TODO: literal view is not allowed in MySQL
```

Comment:
TODO: Lateral view is not allowed in MySQL. I haven't think out corresponding syntax in MySQL. To fulfill this function (e.g.: "select hashtag, count(*) ...") we have to join the result list and return it's count. 
Any suggestions?